server/next.config.js

```diff
@@ -0,0 +1,5 @@
+/** @type {import('next').NextConfig} */
+
+const withTM = require('next-transpile-modules')(['@sonatype/react-shared-components']);
```

Comment:
This is the key thing that next.js apps need to do to use RSC.   By default, next does not apply any sort of webpack/transpilation logic to external dependency code, and so things like image imports and scss imports in RSC cause build errors.  This `next-transpile-modules` plugin tells next to apply transpilation to the speciffied dependency packages. Apparently it has a performance cost, which is why it should only be used on the packages that really need it.

Reply:
Supposedly next 12 uses a rust compiler for everything that's wicked fast, so it might all even out? It'll be interesting to see if that's the case.

Reply:
> Supposedly next 12 uses a rust compiler for everything that's wicked fast, so it might all even out? It'll be interesting to see if that's the case.

I wouldn't even say the configuration seen here is particularly slow.  Just that there's a reason they don't transpile your entire `node_modules` dir by default